lib/lightning_web/channels/attempt_json.ex

```diff
@@ -7,15 +7,16 @@ defmodule LightningWeb.AttemptJson do
   alias Lightning.Workflows.Job
   alias Lightning.Workflows.Trigger
 
-  def render(%Attempt{} = attempt) do
+  def render(%Attempt{} = attempt, include_run_results: include_run_results) do
```

Comment:
i don't love this @midigofrank . we will be passing _lots_ of options to the worker based on project settings. they may include:

1. `include_run_results`
2. `allow_console_log`
3. `other_things_i_cant_remember_right_now`

I wonder if whether or not zero-persistance was enabled should be stamped onto the attempt itself?

Reply:
Good point. I was trying to imitate `render(conn, assigns)`.

The only challenge I can see is that the `Attempt` is the `plan`.
What if we merge them in later?, like:
```elixir
plan = AttemptJson.render(attempt)
Map.merge(plan, opts)
```

Reply:
I'm glad you guys are talking about this. It's a moment where we are departing from Attempt into what the worker refers (at least in spirit) as a Plan.

I'm happy with adding a field to the payload we send, like: 

```
{ ...,  // current "AttemptJson"
  options: { 
    output_dataclips: false | true,
    ... // room for others
  }
}
```

However this is (from a Lightning perspective) not an `Attempt` anymore, even though it's 99% the same.

I would suggest renaming `AttemptJson` to `PlanJson` or `RunPlanJson` (because Attempts are going to be called Runs everywhere soon).

The module would take an Attempt model and any other arguments and return the new payload.

I think @josephjclark should chime in here about potential contract changes; what does a plan look like in the worker? Would it make sense for us to use that word in Lightning given the example above?

Reply:
Well, first off, as Stu says, I expect the "attempt" coming from Lightning to have an `options` key with a bunch of attempt-specific configuration on it.

I am perfectly happy with this setup. I don't care how Lightning assembles this Attempt object, but the JSON that is sent to me right now is convenient, effective and scalable. I'm not in a rush to break that contract.

It's worth noting that I actually internally split the Attempt it into two objects - an ExecutionPlan and Options.

An Attempt/Run is a Lightning abstraction. It doesn't exist in the engine or runtime - although the ExecutionPlan is broadly equivalent.

However, if Lightning had the ability to take a WorkOrder and generate an ExecutionPlan (with `options`), we could a) send that verbatim to the worker, and b) download it locally to run in the CLI (maybe the CLI doesn't need the options?).

I don't care about a) but I think b) is neat, so I would vote in favour.

Reply:
Here is the Attempt -> ExecutionPlan conversion code: https://github.com/OpenFn/kit/blob/main/packages/ws-worker/src/util/convert-attempt.ts

Here is the ExecutionPlan typedef (it's a big fragmented): https://github.com/OpenFn/kit/blob/df38b06b50665131cb2ec0c842af544e6c74ed21/packages/runtime/src/types.ts#L56

And here are the attempt options so far (not much to look at actually): https://github.com/OpenFn/kit/blob/df38b06b50665131cb2ec0c842af544e6c74ed21/packages/ws-worker/src/types.d.ts#L76

Reply:
@taylordowns2000 I've updated the changes